src/System.Management.Automation/engine/runtime/Operations/VariableOps.cs

```diff
@@ -46,6 +46,12 @@ internal static object SetVariableValue(VariablePath variablePath, object value,
                                      : GetAttributeCollection(attributeAsts);
                 var = new PSVariable(variablePath.UnqualifiedPath, value, ScopedItemOptions.None, attributes);
 
+                if (attributes.Count > 0)
+                {
+                    // When there is any attributes, it's possible the value was converted/transformed.
+                    value = var.Value;
```

Comment:
This could potentially trigger a debugger stop since `Value` invokes `DebuggerCheckVariableRead`.

I'm unsure whether that would be desirable, I'm kind of leaning towards yes but it might make variable breakpoints with `ReadWrite` a little awkward. If we do want that behavior we should add an else here that is `else { var.DebuggerCheckVariableRead() }` so it's consistent.

If we don't want that it becomes a little tricky. We may need to add a `PSVariable.GetValueRaw` (there's an existing `SetValueRaw`) which just returns `_value`.

Very clever solution though I thought it was all going to need to be fixed in the code gen ❤️

Reply:
Good point about consistency. I did think about the debugger check for a minute, but didn't think of the consistency issue.
The method `GetValueRaw` is added, please take another look.